### PR TITLE
Call setpgid flag

### PR DESCRIPTION
**Pull request by anti-social**

### CODE REVIEW

File: src/error.rs [hunk 1]
```diff
@@ -17,6 +17,7 @@ pub enum ErrorCode {
     SetUser = 8,
     ChangeRoot = 9,
     SetIdMap = 10,
+    SetPGid = 11,
```
Comment (tailhook) [inline src/error.rs]: I'm not sure we need to send this error code from child. I think we can always to this in parent.

File: src/std_api.rs [hunk 2]
```diff
@@ -151,5 +151,10 @@ impl Command {
         self.config.supplementary_gids = Some(ids);
         self
     }
+
+    pub fn call_setpgid(&mut self, call_setpgid: bool) -> &mut Command {
```
Comment (tailhook) [inline src/std_api.rs]: This should be named `make_group_leader`. And it should be in `std_api` but in `linux` module. The `std_api` is a place where we duplicate the `std::process::Command` API.

Also good docstring should accompany description of the method. With the reason of why would you like to make process a group leader.

**Comment by tailhook** [comment]
Apart from coments in code, I think that group leadership should be contributed here. But the terminal control should be implemented in the vagga itself.

**Comment by anti-social** [comment]
Implemented terminal control in the vagga, all tests passed.

**Comment by anti-social** [comment]
Dont understand how I could rid of `ErrorCode` for setpgid. Why we need `ErrorCode` enum at all? I think there should be only `Error` enum with different `From` implementations.

**Comment by tailhook** [comment]
> Why we need ErrorCode enum at all? 

Error code is sent from child process to parent when eror happens in child. So it is required only for errors that happen in child.

Othewise looks fine. You may squash the commits.

**Comment by anti-social** [comment]
To rid of `ErrorCode::SetPGid` I should write something like:
```
match setpgid(pid, pid) {
    Ok(_) => {},
    Result::Err(nix::Error::Sys(x)) => return Result::Err(Error::SetPGid(x as i32)),
    Result::Err(nix::Error::InvalidPath) => return Result::Err(Error::InvalidPath),
}
```

Also there are another errors with codes that happen only in parent process.

**Comment by tailhook** [comment]
Ah, sorry. I probably misremember something :)

**Pull request merged**